File: backend/src/digitalcard/schemas/operations.py
from pydantic import BaseModel


class OnboardingStep(BaseModel):
    code: str
    name: str
    completed: bool
    path: str


class OnboardingResponse(BaseModel):
    completed: bool
    completed_count: int
    total_count: int
    steps: list[OnboardingStep]


class RateMetric(BaseModel):
    attempts: int
    successes: int
    success_rate: float | None


class MonitoringResponse(BaseModel):
    requests: int
    errors: int
    error_rate: float
    p95_duration_ms: float
    card_publish: RateMetric
    public_card: RateMetric
    lead_submit: RateMetric
    average_first_response_minutes: float | None

File: frontend/admin/src/operations.ts
export interface OnboardingStatus { completed: boolean; completed_count: number; total_count: number; steps: Array<{ code: string; name: string; completed: boolean; path: string }> }
export interface RateMetric { attempts: number; successes: number; success_rate: number | null }
export interface Monitoring { requests: number; errors: number; error_rate: number; p95_duration_ms: number; card_publish: RateMetric; public_card: RateMetric; lead_submit: RateMetric; average_first_response_minutes: number | null }
